ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -171,32 +161,40 @@
         .withClientSslSupport();
   }
 
-  private static Map<String, Object> commonConfigs(Map<String, Object> props) {
-    return props.entrySet().stream()
-        .filter(e -> !e.getKey().startsWith(KSQL_CONFIG_PROPERTY_PREFIX))
-        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  private static void applyPrefixedConfig(final Map<String, Object> props,
+                                          final String prefix,
+                                          final String key,
+                                          final Object value) {
+    props.put(
+        key.startsWith(prefix) ? key.substring(prefix.length()) : key,
+        value
+    );
   }
 
-  private static Map<String, Object> propertiesWithPrefix(
-      Map<String, Object> props, String prefix) {
-    AbstractConfig abstractConfig = new AbstractConfig(new ConfigDef(), props);
-    return abstractConfig.originalsWithPrefix(prefix);
+  private static boolean streamsConfigFilter(final String config) {
+    if (config.startsWith(KSQL_STREAMS_PREFIX)) {
+      return true;
+    }
+    return StreamsConfig.configDef().names().contains(config)
+        || ConsumerConfig.configNames().contains(config)
+        || ProducerConfig.configNames().contains(config)
+        || AdminClientConfig.configNames().contains(config);
   }
 
-
-  private void applyStreamsConfig(Map<String, Object> props) {
-    ksqlStreamConfigProps.putAll(commonConfigs(props));
-    ksqlStreamConfigProps.putAll(propertiesWithPrefix(props, KSQL_STREAMS_PREFIX));
+  private void applyStreamsConfig(final Map<String, Object> props) {
```

Comment:
Isn't this doing something very similar to `originalsWithPrefix `?  Would be good to make use of this rather than reinvent the wheel.

Reply:
This method filters streams configs by looking in the streams/consumer/producer config-defs, not just by prefix.